x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorCustomSchedule.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class ConnectorCustomSchedule implements Writeable, ToXContentObject {
+
+    private final ConfigurationOverrides configurationOverrides;
+    private final Boolean enabled;
+    private final String interval;
+    @Nullable
+    private final String lastSynced;
```

Comment:
<img width="548" alt="Screenshot 2023-11-22 at 16 02 17" src="https://github.com/elastic/elasticsearch/assets/14121688/4bfc7c8b-40a9-4bb5-b570-e29d352e5d38">

According to [Elastic docs](https://www.elastic.co/guide/en/enterprise-search/8.9/crawler-managing.html#crawler-managing-multiple-schedules) it's a String (?) 

But I see that for [Filtering we should be handling data as Dates ](https://github.com/elastic/elasticsearch/blob/05b2591d0342d9735fd5f88bffe741ffd83a9607/x-pack/plugin/core/template-resources/src/main/resources/entsearch/connector/elastic-connectors-mappings.json#L74C20-L76C23) I will fix it there

Reply:
I took a look at [CONNECTOR_PROTOCOL.md](https://github.com/elastic/connectors/blob/main/docs/CONNECTOR_PROTOCOL.md) search for "advanced snippets", fields like `created_at` or `updated_at` are expected to be a string

```
                "advanced_snippet": {
                  "created_at": "2023-11-09T15:13:08.231Z",
                  "updated_at": "2023-11-09T15:13:08.231Z",
                  "value": {}
                },
```
I think it can stay as a `String`

Reply:
Ah, it's a string there because Elasticsearch automatically parses date strings. It makes more sense to have it represented as a date internally.

Reply:
but it's not a String, right? It's a timestamp. Don't we want to parse that to ensure it's correctly formatted, and create the actual timestamp on the server side?

Reply:
Yes exactly @carlosdelest. The difference is that the connector protocol was an external represenation, and this is an internal representation, so these should be dates.

Reply:
It seems that `Instant` class is widely used in the codebase to handle time/dates, I will use this to represent time related data